### PR TITLE
Migrate to NeoVim using Lua

### PR DESCRIPTION
**Pull request by fatih**
This is a WIP project.

I'll try to migrate my `vimrc` to a `init.lua`. Instead of copy pasting from other dotfiles, my plan is slowly my own configuration, so I can understand how Lua configuration works within NeoVim.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -0,0 +1,397 @@
+local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
```
Comment (dogukanayd) [inline init.lua:1]: I apologize if this is a silly question, but how will you handle the faith/vim-go package with this structure?

Comment (fatih) [reply init.lua:1]: I replaced most of vim-go functionality with lsp, treesitter. I might extract some functionality from vim-go into nvim specific plugins, but didn't decided yet.

**Comment by fatih** [comment]
```
alias vim=nvim
```

File: alacritty.yml [hunk 2]
```diff
@@ -58,17 +58,17 @@ scrolling:
...
   # The bold font face
   bold:
-    family: Berkeley Mono
+    family: BerkeleyMono Nerd Font Mono
     style: Bold
 
   # The italic font face
   italic:
-    family: Berkeley Mono
+    family: BerkeleyMono Nerd Font Mono
```
Comment (mrnugget) [inline alacritty.yml:71]: :sunglasses:

Comment (fatih) [reply alacritty.yml:71]: @mrnugget your comment about "emoji" changed my mind :) There are some icons I dislike, but I'll try it out at least for couple of weeks.

Comment (mrnugget) [reply alacritty.yml:71]: Yeah, agree. It somehow also doesn't work as well on macOS as on Linux for me.

**Pull request merged**